src/lib/components/constrainedtext/Constrainedtext.component.js

```diff
@@ -35,27 +35,48 @@ const BlockTooltip = styled.div`
   }
 `;
 
+function isEllipsisActive(element) {
+  return element && (element.offsetWidth < element.scrollWidth);
+}
+
+function getConstrainedTextContainer(constrainedTextRef, lineClamp, text) {
+  return <ConstrainedTextContainer
+      ref={constrainedTextRef}
+      className="sc-constrainedtext"
+      lineClamp={lineClamp}
+  >
+    {text}
+  </ConstrainedTextContainer>;
+}
+
 function ConstrainedText({
-  text,
-  tooltipStyle,
-  tooltipPlacement,
-  lineClamp = 1,
-}: Props): Node {
+                           text,
+                           tooltipStyle,
+                           tooltipPlacement,
+                           lineClamp = 1,
+                         }: Props): Node {
+
+  const constrainedTextRef = useRef<HTMLDivElement | null>(null);
+  const [displayToolTip,setDisplayToolTip] = useState(false);
+
+  useEffect(() => {
+    setDisplayToolTip(isEllipsisActive(constrainedTextRef.current) ||  lineClamp > 1);
+  }, []);
```

Comment:
I would refactor it in this way:

```suggestion
  const [displayToolTip,setDisplayToolTip] = useState(false);

  const constrainedTextRef = useCallback((element) => {
    setDisplayToolTip(isEllipsisActive(element) ||  lineClamp > 1);
  }, [lineClamp]);
```